workflow-bot-app/src/main/java/com/symphony/bdk/workflow/engine/camunda/monitoring/converter/WorkflowInstDomainVersionConverter.java

```diff
@@ -0,0 +1,25 @@
+package com.symphony.bdk.workflow.engine.camunda.monitoring.converter;
+
+import com.symphony.bdk.workflow.converter.BiConverter;
+import com.symphony.bdk.workflow.monitoring.repository.domain.WorkflowInstanceDomain;
+
+import org.camunda.bpm.engine.impl.persistence.entity.HistoricProcessInstanceEntity;
+import org.springframework.stereotype.Component;
+
+import java.util.Map;
+import java.util.Optional;
+
+@Component
+public class WorkflowInstDomainVersionConverter extends AbstractInstanceDomainConverter
+    implements BiConverter<HistoricProcessInstanceEntity, Map<String, String>, WorkflowInstanceDomain> {
+
+  @Override
+  public WorkflowInstanceDomain apply(HistoricProcessInstanceEntity hisProcInstance,
+      Map<String, String> procIdVersionTagMap) {
+    if (!procIdVersionTagMap.containsKey(hisProcInstance.getProcessDefinitionId())) {
+      return null;
+    }
+    Optional<String> version = Optional.ofNullable(procIdVersionTagMap.get(hisProcInstance.getProcessDefinitionId()));
+    return this.instanceCommonBuilder(hisProcInstance).version(version.map(Long::valueOf).orElse(null)).build();
```

Comment:
```suggestion
Optional<String> version = Optional.ofNullable(procIdVersionTagMap.get(hisProcInstance.getProcessDefinitionId()));

if (version.isPresent()) {
  return this.instanceCommonBuilder(hisProcInstance).version(version.map(Long::valueOf).orElse(null)).build();
} else {
  return null;
}
```

Reply:
Overlooked this refactoring.
code changed now